### PR TITLE
fix: price of reward token is zero if it is zero address

### PR DESCRIPTION
**Pull request by viet-nv**

### CODE REVIEW

File: src/hooks/useTokensMarketPrice.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ export default function useTokensMarketPrice(tokens: (Token | null | undefined)[
 
   const tokenAddress = tokens
     .filter(Boolean)
-    .map(token => (token?.isNative ? WETH[chainId || ChainId.MAINNET].address : token?.address))
+    .map(token =>
+      token?.isNative || token?.address === ZERO_ADDRESS ? WETH[chainId || ChainId.MAINNET].address : token?.address,
```
Comment (namgold) [inline src/hooks/useTokensMarketPrice.ts:48]: Why do we need both `token?.isNative` and `token?.address === ZERO_ADDRESS`. Doesnt they are same and always go together?

Comment (viet-nv) [reply src/hooks/useTokensMarketPrice.ts:48]: native token doesnt have address

Comment (viet-nv) [reply src/hooks/useTokensMarketPrice.ts:48]: zero address is only a trick for reward token of farm

File: src/state/application/hooks.ts [hunk 2]
```diff
@@ -365,7 +365,10 @@ export function useTokensPrice(tokens: (Token | NativeCurrency | null | undefine
         const key = `${token.address}_${chainId}_${version}`
         if (cache[key]) return cache[key]
 
-        const tokenPriceByETH = await getTokenPriceByETH(token?.address, client)
+        const tokenPriceByETH = await getTokenPriceByETH(
+          token?.address === ZERO_ADDRESS ? WETH[chainId as ChainId].address : token?.address,
+          client,
+        )
```
Comment (namgold) [inline src/state/application/hooks.ts:371]: Does `getTokenPriceByETH(WETH[chainId as ChainId].address)` always return 1?
If it does, we should simplify these line to `token?.address === ZERO_ADDRESS ? 1 : await ...`

Comment (viet-nv) [reply src/state/application/hooks.ts:371]: LOL, just a special case, we have cache for that so It's only called 1 time, I think we dont need to save 1 request, haha

Comment (namgold) [reply src/state/application/hooks.ts:371]: No bro, it helps much easier to understand the logic to maintain

Comment (viet-nv) [reply src/state/application/hooks.ts]: Make sense. moved it to condition above.

File: src/hooks/useTokensMarketPrice.ts (hunk 1 above)
Comment (namgold) [reply src/hooks/useTokensMarketPrice.ts:48]: These trick logic make it alot harder to maintain, should reduce them as much as possible in the future.

**Pull request merged**